source/StepBro.Core/Parser/StepBroListener.Procedure.cs

```diff
@@ -444,12 +446,264 @@ public override void EnterForStatement([NotNull] SBP.ForStatementContext context
         {
             this.AddEnterStatement(context);
             //m_lastPropertyBlock = null;
-            m_expressionData.PushStackLevel("ForStatement");
             m_enteredLoopStatement = true;
+            m_expressionData.PushStackLevel("for-init");
+            m_scopeStack.Push(new ProcedureParsingScope(m_scopeStack.Peek(), "for-init", ProcedureParsingScope.ScopeType.Block));
+            m_forInitVariables.Push(new List<Expression>());
+        }
+
+        public override void ExitForVariableDeclaration([NotNull] SBP.ForVariableDeclarationContext context)
+        {
+            base.ExitForVariableDeclaration(context);
```

Comment:
I don't believe this is necessary; there's no useful code in the base implementation.